src/basic/log.c

```diff
@@ -1396,12 +1396,12 @@ int log_dup_console(void) {
         return 0;
 }
 
-void log_setup_service(void) {
+void log_setup_service(bool log_to_console) {
         /* Sets up logging the way it is most appropriate for running a program as a service. Note that using this
          * doesn't make the binary unsuitable for invocation on the command line, as log output will still go to the
          * terminal if invoked interactively. */
 
-        log_set_target(LOG_TARGET_AUTO);
+        log_set_target(log_to_console ? LOG_TARGET_CONSOLE : LOG_TARGET_AUTO);
```

Comment:
Hmm, maybe we should combine this with an explicit setting in the units where the binaries that set log_to_console=true are used? We still want to log to the journal from systemd-tmpfiles.service, systemd-sysusers.service, etc.

Reply:
Is there a big difference between logging directly to the journal and having output pass via the console? It ends up in the journal regardless no?

Reply:
Setting the SYSTEMD_LOG_TARGET environment variable in the relevant units works but is a little bit ugly. Maybe there's a way to detect whether we're running as a service and switch between console and journal based on that?

Reply:
Yes, there's a difference: when going via the console, all metadata is stripped (or rather, it is never attached in the first place). I agree that setting of a variable is ugly. Indeed, maybe it'd be better to look at $JOURNAL_STREAM? @poettering, opinions?